rakelib/pg_connectivity_check.rake

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+namespace :db do
+  desc 'Check database connectivity and perform test operations'
+  task check: :environment do
+    puts '*** Test 1: Access Records'
+    begin
+      if UserAccount.exists?
+        puts 'UserAccount records exist.'
+      else
+        puts 'No UserAccount records found but database is up.'
+      end
+    rescue => e
+      Rails.logger.error(
+        { message: "Failed to access UserAccount records during Postgres connectivity check. Error: #{e.message}" }
+      )
+      return # Exit the task if connectivity issues are detected
+    end
+
+    puts '*** Test 2: Perform GET and PUT operations'
+    begin
+      UserAccount.transaction do
+        icn = 'abc123'
+        UserAccount.create!(icn:)
+        puts 'Created UserAccount'
+
+        user = UserAccount.find_by(icn:)
+        puts 'Updated UserAccount ICN' if user.update(icn: 'xyz789')
+
+        # Roll back transaction so the changes aren't persisted.
+        raise ActiveRecord::Rollback
+      end
+      puts 'Transaction rolled back successfully.'
+    rescue => e
+      Rails.logger.error(
+        { message: "Error during GET and PUT operations for Postgres connectivity check. Error: #{e.message}" }
+      )
+      return # Exit the task if connectivity issues are detected
+    end
+
+    puts '*** Test 3: Check Connection'
+    # Order matters. This command returns false if it's run before the other tests
+    if ActiveRecord::Base.connection.active?
+      puts 'Connection is active.'
+    else
+      Rails.logger.error({ message: 'Postgres Connection is NOT active.' })
+      return # Exit the task if connection fails
```

Comment:
I'm `return`ing out of each of these three tests if there is a failure in any of them. Is that ideal or should I let it run through the end and fail on the last check if it fails?

Reply:
Same comment as above 'Postgres Connection is NOT active.' could freak out some people looking at the logs

Reply:
I would show everything as a long and be more descriptive to avoid confusion. 
ex: `puts 'Testing ArgoCD Upgrade - Postgres Connection is NOT active.'`

Reply:
I would not return. Let the task finish. If the connection succeeds later on, it would be nice to know.

Reply:
Good call. I updated the message.